src/multi_vector_simulator/C1_verification.py

```diff
@@ -766,3 +770,87 @@ def check_for_sufficient_assets_on_busses(dict_values):
                 f"so that the bus is not a dead end should be two, excluding the excess sink. "
                 f"These are the connected assets: {asset_string}"
             )
+
+
+def check_energy_system_can_fulfill_max_demand(dict_values):
+    r"""
+    Helps to do oemof-solph termination debugging: Logs a logging.warning message if the aggregated installed capacity and maximum capacity (if applicable)
+    of all conversion, generation and storage assets connected to one bus is smaller than the maximum demand.
+
+   The check is applied to each bus of the energy system. Check passes when the potential peak supply is
+   larger then or equal to the peak demand on the bus, or if the maximum capacity of an asset is set to
+   None when optimizing.
+
+    Parameters
+    ----------
+    dict_values : dict
+        Contains all input data of the simulation.
+
+    Returns
+    -------
+    Indirectly, logs a logging.warning message if the installed and maximum capacities of
+    conversion/generation/storage assets are less than the maximum demand, for each bus.
+
+    Notes
+    -----
+    Tested with:
+    - test_check_energy_system_can_fulfill_max_demand_sufficient_capacities()
+    - test_check_energy_system_can_fulfill_max_demand_no_maximum_capacity()
+    - test_check_energy_system_can_fulfill_max_demand_insufficient_capacities()
+    - test_check_energy_system_can_fulfill_max_demand_with_storage()
+    """
+    for bus in dict_values[ENERGY_BUSSES]:
+        pass_check = False
+        opt_cap_storage = False
+        peak_demand = 0
+        peak_generation = 0
+        for item in dict_values[ENERGY_BUSSES][bus][ASSET_DICT]:
+            # filters out excess energy sinks, leaving only actual demand profiles
+            if item in dict_values[ENERGY_CONSUMPTION]:
+                if DISPATCHABILITY in dict_values[ENERGY_CONSUMPTION][item]:
```

Comment:
I think so, because before it was:

if item in dict_values[ENERGY_CONSUMPTION] and dict_values[ENERGY_CONSUMPTION][item][DISPATCHABILITY] is False

But the Dispatchability key doesn't exist in the demand dictionary, so this would raise an error. So it should first be checked that the key exists in the dictionary

Reply:
Ahh, yes, you`re right. We don´t add it [here](https://github.com/rl-institut/multi-vector-simulator/blob/feature/energy_system_max_demand_fulfillment/src/multi_vector_simulator/C0_data_processing.py#L429)

But... then... that might be the reason why I can not reproduce the warning using input data? If demands never have the `DISPATCHABILITY` key, they will never be added to `peak_demand` and there never is a warning. 

I think you should add the key somewhere in the above function (if `FILENAME` is [given](https://github.com/rl-institut/multi-vector-simulator/blob/feature/energy_system_max_demand_fulfillment/src/multi_vector_simulator/C0_data_processing.py#L448)). 

Alternatively, you can filter the excess sinks out. The bus processing results in:
```
        "ESS Li-Ion bus": {
            "Asset_list": {
                "Charge_contoller_ESS_Li-Ion_(charge)": "Charge_contoller_ESS_Li-Ion_(charge)",
                "Charge_contoller_ESS_Li-Ion_(discharge)": "Charge_contoller_ESS_Li-Ion_(discharge)",
                "ESS Li-Ion bus_excess": "ESS Li-Ion bus_excess_sink",
                "storage_01": "ESS Li-Ion"
            },
            "_excess": "ESS Li-Ion bus_excess",
            "energyVector": "Electricity",
            "label": "ESS Li-Ion bus"
```
So if you check that, in this case, `item != dict_values[busses]["ESS Li-Ion bus"][ "_excess"]` you should avoid the excess sinks (global variable `EXCESS="_excess"`)

Reply:
Basically, what we now are going though is when we use very complex tests and think that our pytests only are enough to check their function. As you see, they may not. So, one should always try to provoke the warning with changing the input files of the MVS. If necessary and if a lot of things dont add up, it may even be possible to add a small little benchmark test for this.